Adding CountDownActivity, MainActivity and StartTimerActivity

hanandarugar opened the pull request.

hanandarugar committed:
Adding CountDownActivity, MainActivity and StartTimerActivity

Vardaansh1 commented on app/src/main/java/com/kaamchor/android/fameofshame/CountDownActivity.java:
```diff
@@ -0,0 +1,40 @@
+public class CountDownActivity extends AppCompatActivity {
+
+    int hr = 0;
+    int min = 0;
```
private as we should always keep visibility to the lowest possible as general practice.

Vardaansh1 commented on app/src/main/java/com/kaamchor/android/fameofshame/CountDownActivity.java:
```diff
@@ -0,0 +1,40 @@
+        min = mins.getValue();
+
+    }
+     public int getHour(){
```
is this used anywhere?

Vardaansh1 commented on app/src/main/java/com/kaamchor/android/fameofshame/CountDownActivity.java:
```diff
@@ -0,0 +1,40 @@
+     public int getHour(){
+        return hr;
+     }
+     public int getMin(){
```
same

Vardaansh1 commented on app/src/main/java/com/kaamchor/android/fameofshame/CountDownActivity.java:
```diff
@@ -0,0 +1,40 @@
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_count_down);
+        NumberPicker hour = (NumberPicker) findViewById(R.id.hour);
```
you can probably make the UI objects global, and get a reference to them in onCreate like you have done here.
This way if you have to use the ui element at some other place, you wont have to keep getting a reference to it everytime

Vardaansh1 commented on app/src/main/java/com/kaamchor/android/fameofshame/StartTimerActivity.java:
```diff
@@ -0,0 +1,47 @@
+        countDownStart();
+    }
+
+    public void initTimer(){
```
private

Vardaansh1 commented on app/src/main/java/com/kaamchor/android/fameofshame/StartTimerActivity.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public void countDownStart() {
+        int hr = getHour();
```
instead of having getters, pass the value in the intent while starting the activity

Vardaansh1 commented on app/src/main/java/com/kaamchor/android/fameofshame/StartTimerActivity.java:
```diff
@@ -0,0 +1,47 @@
+ */
+
+
+public class StartTimerActivity extends CountDownActivity {
```
why extend CountDownActivity?

Vardaansh1 reviewed (commented):
minor comments.

hanandarugar committed:
The Count down timer is working.
When Stop button is pressed in the CountDownActivity the App direct to the mainActivity.
When Reset button is pressed in the CountDownActivity the App directs to the StartTimerActivity.
Added new screen once the user finishes the set time. Added FinishActivity to get make a call to the finish screen and display "Well Done!!" message on the screen.

To Do: Add a toast when user tries to start timer at 00:00.

hanandarugar committed:
Added toast when user click start while timer is at 00:00.

hanandarugar committed:
1. Fixed timer issue to pause the timer when an incomming call is received and resume the timer after the call is done.
2. Fixed timer issue to keep the screen awake in order to prevent timer from stopping.
3. Made UI changes.

hanandarugar committed:
Changed background color of finish lauout and start timer layout

hanandarugar committed:
Adding CountDownActivity, MainActivity and StartTimerActivity
